src/commands/add.ts

```diff
@@ -61,10 +61,13 @@ export default defineCommand({
     const config = await kit.loadNuxtConfig({ cwd })
 
     // Resolve template
-    const res = template({ name, args: ctx.args })
+    const res = template({ name, args: ctx.args, nuxt: config })
+
+    // Change resolution root dir when applicable
+    const resolveFromRoot = config.future.compatibilityVersion === 4 && templateName === 'api'
 
     // Resolve full path to generated file
-    const path = resolve(config.srcDir, res.path)
+    const path = resolve(resolveFromRoot ? config.rootDir : config.srcDir, res.path)
```

Comment:
I think we can safely always resolve from `config.srcDir`, even without `compatibilityVersion: 4`, as it's always been possible to set a custom `srcDir`...

The only exception is modules, server, etc. But this bug _already_ exists and probably needs another solution (e.g. specifying in template somehow whether it is relative to srcDir or rootDir.

Reply:
I've changed it now so that the full path is resolved from within the template function. I suppose this will be easier to expand if we will add templates for the other exceptions you mention.